plugins/restful/RestfulEntityBase.php

```diff
@@ -689,8 +768,48 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $null_missi
         continue;
       }
 
-      $field_value = $this->propertyValuesPreprocess($property_name, $request[$public_field_name], $public_field_name);
-      $wrapper->{$property_name}->set($field_value);
+      if (field_is_translatable($this->entityType, field_info_field($info['property']))) {
+        $values = $request[$public_field_name];
+
+        // Parse values in case it still not an array.
+        if (!is_array($values)) {
+          $values = drupal_json_decode($values);
```

Comment:
but I agree the drupal_json_decode() looks weird, and should probably be _not_ needed.